business/dashboards.go

```diff
@@ -39,12 +39,28 @@ func NewDashboardsService() *DashboardsService {
 	if nsLabel == "" {
 		nsLabel = "kubernetes_namespace"
 	}
+
+	// Overwrite Custom dashboards defined at Namespace level
+	builtInDashboards := cfg.CustomDashboards.OrganizeByName()
+	if namespace != nil {
+		nsDashboards := dashboards.GetNamespaceMonitoringDashboards(namespace.Name, namespace.Annotations)
+		for name, dashboard := range nsDashboards.OrganizeByName() {
+			builtInDashboards[name] = dashboard
+		}
+	}
+	if workload != nil {
+		wkDashboards := dashboards.GetWorkloadMonitoringDashboards(namespace.Name, workload.Name, workload.DashboardAnnotations)
+		for name, dashboard := range wkDashboards.OrganizeByName() {
+			builtInDashboards[name] = dashboard
+		}
+	}
+
 	return &DashboardsService{
 		CustomEnabled:   customEnabled,
 		promConfig:      prom,
 		globalNamespace: cfg.Deployment.Namespace,
 		namespaceLabel:  nsLabel,
-		dashboards:      cfg.CustomDashboards.OrganizeByName(),
+		dashboards:      builtInDashboards,
```

Comment:
See previous comment - this change goes with the bigger change in the earlier suggested change
```suggestion
		dashboards:      builtInDashboards.OrganizeByName(),
```